routing_model/src/scenario_tests.rs

```diff
@@ -825,6 +825,110 @@ mod dst_tests {
             &AssertState::default(),
         );
     }
+
+    #[test]
+    fn test_local_event_offline() {
+        run_test(
+            "Get local event node detected offline",
+            &intial_state_old_elders(),
+            &[LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_votes: vec![ParsecVote::Offline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_online_again() {
+        run_test(
+            "Get local event node detected online again",
+            &intial_state_old_elders(),
+            &[LocalEvent::NodeDetectedBackOnline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_votes: vec![ParsecVote::BackOnline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_then_parsec_offline() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event()],
+        );
+        run_test(
+            "Get local event node detected offline and then parsec consensus offline",
+            &initial_state,
+            &[ParsecVote::Offline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_nodes: vec![NodeChange::Offline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_and_parsec_offline_then_check_elder() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[
+                LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event(),
+                ParsecVote::Offline(NODE_ELDER_130).to_event(),
+            ],
+        );
+        run_test(
+            "Get parsec consensus offline then check elder",
+            &initial_state,
+            &[ParsecVote::CheckElder.to_event()],
+            &AssertState {
+                action_our_events: vec![LocalEvent::TimeoutCheckElder],
+                ..AssertState::default()
+            },
```

Comment:
That test seem to be failing, something in CheckElder does not take into account the offline state.
I think this is part of this PR's job.

Reply:
Interesting, ok I'll have a look!

Reply:
To clarify, what I meant is I expect the assert state to be different and have votes for new elders.

Reply:
It may be this test would be better grouped with the other check elder tests. Maybe.